docker-compose.yml

```diff
@@ -68,6 +68,7 @@ services:
       volumes:
           - postgres-data:/var/lib/postgresql/data
           - postgres-log:/var/log/postgresql
+          - ./arches/install/init.sql:/docker-entrypoint-initdb.d/init.sql # to set up the DB template
```

Comment:
Not sure if this is the right solution, but I may be missing something (it's late here). What if you have a database running in aws? Would you need to manually run this script on that db before you can let the Arches container run setup_db? Could this script be incorporated in setup_db instead?

Reply:
This is letting the postgres container set up the template that Arches projects use to base themselves on. If the script was in setup_db, then the Arches container would have to have superuser postgresql credentials to be able to add the two extensions, which is something that should only happen once. For context, by IT policy, no app will ever get that sort of level of access to the provisioned DB. Having the required template be installed as part of deployment/provisioning, rather than when an app is being set up is not a big change, but is pretty crucial in a shared environment.

A fallback could be the entrypoint.sh looking for superuser creds in the environment variables, and running the sql script if they are present?

Reply:
This addition to docker-compose is to allow for local deployment with pg in a 'nearby' container under the control of docker-compose.

Reply:
Got it. Not forcing the use of superuser credentials seems a good move to me.

I'm still not entirely clear on how the deployment process will go on a non-local environment. As far as I can remember, currently when running Arches in a clean environment with an empty, newly provisioned Postgres db, all the user needs to do is run the Arches container and the db will be setup in a completely automated way. Would we lose this convenience with this new Postgis extensions script? 

I do think your idea of looking for superuser credentials and running the script sounds good, so that at least we have a simpler (yet less secure) way to get people started. Could that be incorporated into this PR?

By the way, I do not see any directions regarding this provisioning script in the regular installation manual:
https://arches.readthedocs.io/en/stable/installation/
Should this be mentioned there?

Reply:
Hi @veuncent , yes, it should definitely be mentioned there. I actually have a whole lot of install-related docs tickets grouped together in a project because I think a big overhaul of that process is warranted at this point. https://github.com/archesproject/arches-docs/projects/1

However, this change is not deployed yet in any published release, so if anything it should just be in the `latest` docs, not the `master` ones.